### PR TITLE
feat(balance): Advanced Microreactor No Longer Irradiates You

### PR DESCRIPTION
**Pull request by yay855**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [ ] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

The advanced microreactor is described as being safer than the regular microreactor, and while this is technically true, it's just 2/3rds less radioactive; it's still plenty dangerous to anyone without a radiation scrubber CBM. And given how rare both it and its prerequisite is, and how difficult it can be to get them AND a radiation scrubber CBM, I don't think it's unreasonable to just make the advanced microreactor not irradiate you.

## Describe the solution

A single number in the C++ was changed, effectively eliminating the radiation output from the advanced microreactor.

### CODE REVIEW

File: src/suffer.cpp [hunk 1]
```diff
@@ -1240,7 +1240,7 @@ void Character::suffer_from_radiation()
         if( calendar::once_every( 60_minutes ) ) {
             int rad_mod = 0;
             rad_mod += has_bionic( bio_reactor ) ? 3 : 0;
-            rad_mod += has_bionic( bio_advreactor ) ? 2 : 0;
+            rad_mod += has_bionic( bio_advreactor ) ? 0 : 0;
```
Comment (chaosvolt) [inline src/suffer.cpp:1243]: Would this technically mean that omitting the line entirely would be the preferred way to do this?

Comment (yay855) [reply src/suffer.cpp:1243]: Huh, good question. I'll check in on it tomorrow and see if it causes any errors.

**Comment by RoyalFox2140** [comment]
image
Well I can't speak for code reviewing it but it works.

**Review by RoyalFox2140 — APPROVED** [review]
Works perfectly. Compile tested on MSYS2. Thanks Chorus, I finally used CBM's again after 6 years because of this. :D

File: src/suffer.cpp (hunk 1 above)
Comment (scarf005) [reply src/suffer.cpp:1243]: yes, omitting the line will work without issues.

Comment (scarf005) [inline src/suffer.cpp:1243]:
```suggestion
```

**Review by scarf005 — COMMENTED** [review]
maybe we should consult this change with @MrLostman who came up with the initial idea

**Review by scarf005 — CHANGES_REQUESTED** [review]
maybe we should consult this change with @MrLostman who came up with the initial idea

**Comment by RoyalFox2140** [comment]
Can always consult them, my testing and approval stands so long as the function of the... function isn't altered. Someone else can merge.

**Review by scarf005 — APPROVED** [review]

**Comment by KheirFerrum** [comment]
I don't agree with this. The entire point of them is that they require the scrubber to function at their safest and max capacity.

Last I checked, the advanced microreactor doesn't even produce radiation unless you insert plutonium into it and activate the override.

**Comment by chaosvolt** [comment]
> Last I checked, the advanced microreactor doesn't even produce radiation unless you insert plutonium into it and activate the override.

Incorrect, it would trigger constant radiation so long as you have the bionic installed at all, so long as any fuel is loaded:
image

It's also a whopping 1 less radiation per hour compared to the regular reactor, so the difference in effective safety was rather underwhelming.

**Pull request merged**